### PR TITLE
removed all references of a supplier. fix #247

### LINKED ISSUE #247: remove any reference of 'supplier'
The code has references to a _supplier_ property for models such as _Dataset_.
It should be removed from the code as well as other references to it.

### PR DESCRIPTION
**Pull request by vinyll**
Let's first see how Jenkins react.
The API was affected removing the field _supplier_. Shouldn't we increment the API version?

### CODE REVIEW

**Comment by davidbgk** [comment]
I relaunched the Travis build, not sure the error was related to the modification.
About the API, we don't have such policy yet but that might be a good idea, @noirbizarre?

**Comment by davidbgk** [comment]
r+

File: .gitignore [hunk 1]
```diff
@@ -38,6 +38,7 @@ js-built
 /node_modules/
 /udata/static/
 /jsdoc/
+dump.rdb
```
Comment (noirbizarre) [inline .gitignore]: *.rdb ?

Comment (davidbgk) [inline .gitignore]: redis, if launched in the same folder.

File: udata/migrations/2015-11-16-remove-supplier.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Remove supplier property from datasets
+ */
+function process_collection(collection) {
+    collection.update({supplier: {$exists: true}},
+                                 {$unset: {'supplier': 1}}, false, true);
```
Comment (noirbizarre) [inline udata/migrations/2015-11-16-remove-supplier.js]: It should be indented with `{supplier` instead of `{$exists`

File: .gitignore (hunk 1 above)
Comment (vinyll) [inline .gitignore]: The demand is to wildcard rdb extensions, right?

File: udata/migrations/2015-11-16-remove-supplier.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    var count = db.runCommand({getLastError : 1}).n;
```
Comment (noirbizarre) [inline udata/migrations/2015-11-16-remove-supplier.js]: You can also use the update return value which contain `nModified`. The current pattern will only work if there was no other operation in the interval.

File: .gitignore (hunk 1 above)
Comment (noirbizarre) [inline .gitignore]: Exactly ;)

**Comment by noirbizarre** [comment]
r+wc

**Pull request merged**